From f5dbd1fdd5bf0a85bda432066855f563105fbf9d Mon Sep 17 00:00:00 2001
From: Henning Schild <henning.schild@siemens.com>
Date: Wed, 27 Sep 2017 12:59:49 +0200
Subject: [PATCH] ivshmem-net: add MAC changing interface

Allow ifconfig, ip and other such tools to change the MAC of the
virtual NIC.

Signed-off-by: Henning Schild <henning.schild@siemens.com>
Signed-off-by: Jan Kiszka <jan.kiszka@siemens.com>
(cherry picked from commit 7744d59d11101f75afaa6f550cabba38c2c0d260)
---
 drivers/net/ivshmem-net.c | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

--- a/drivers/net/ivshmem-net.c
+++ b/drivers/net/ivshmem-net.c
@@ -685,12 +685,14 @@ static void ivshm_net_poll_controller(st
 #endif
 
 static const struct net_device_ops ivshm_net_ops = {
-	.ndo_open	= ivshm_net_open,
-	.ndo_stop	= ivshm_net_stop,
-	.ndo_start_xmit	= ivshm_net_xmit,
-	.ndo_change_mtu	= ivshm_net_change_mtu,
+	.ndo_open		= ivshm_net_open,
+	.ndo_stop		= ivshm_net_stop,
+	.ndo_start_xmit		= ivshm_net_xmit,
+	.ndo_change_mtu		= ivshm_net_change_mtu,
+	.ndo_set_mac_address 	= eth_mac_addr,
+	.ndo_validate_addr	= eth_validate_addr,
 #ifdef CONFIG_NET_POLL_CONTROLLER
-	.ndo_poll_controller = ivshm_net_poll_controller,
+	.ndo_poll_controller	= ivshm_net_poll_controller,
 #endif
 };
 
